src/screens/Accounts/Accounts.js

```diff
@@ -295,40 +312,114 @@ class AccountsScreen extends React.Component<Props, State> {
         )}
       />
     );
+  }
+
+  renderInitializeAccount(item: NewItem) {
+    const { baseFiatCurrency } = this.props;
+
+    const { action } = item;
+    const fiatCurrency = baseFiatCurrency || defaultFiatCurrency;
+    const currencySymbol = getCurrencySymbol(fiatCurrency);
+
+    return (
+      <SettingsItemCarded
+        title="Ethereum Smart Wallet"
+        subtitle={`${currencySymbol} 0`}
+        onMainPress={() => action()}
+        onSettingsPress={null}
+        isActive={false}
+        customIcon={(
+          <IconWrapper>
+            <WalletIcon isSmart />
+          </IconWrapper>
+        )}
+      />
+    );
+  }
+
+  renderListItem = ({ item }: ListElement) => {
+    switch (item.type) {
+      case 'NETWORK':
+        return this.renderNetwork(item);
+
+      case 'ACCOUNT':
+        return this.renderAccount(item);
+
+      case 'NEW_SMART_WALLET':
+        return this.renderInitializeAccount(item);
+
+      default:
+        return null;
+    }
   };
 
-  render() {
-    const { showPinModal, changingAccount } = this.state;
+  visibleAccounts(accounts: Accounts, smartWalletFeatureEnabled: boolean): Accounts {
+    if (smartWalletFeatureEnabled) {
+      return accounts;
+    }
+
+    return accounts.filter(({ type }) => type !== ACCOUNT_TYPES.SMART_WALLET);
+  }
+
+  wallets(): ListItem[] {
     const {
       accounts,
-      blockchainNetworks,
-      smartWalletFeatureEnabled,
-      smartWalletState,
       navigation,
       balances,
       assets,
       rates,
+      smartWalletFeatureEnabled,
+      blockchainNetworks,
+      baseFiatCurrency,
     } = this.props;
-    const ppnNetwork = blockchainNetworks.find((network) => network.id === BLOCKCHAIN_NETWORK_TYPES.PILLAR_NETWORK);
-    const smartWalletStatus: SmartWalletStatus = getSmartWalletStatus(accounts, smartWalletState);
-    const showSmartWalletInitButton = !smartWalletStatus.hasAccount && smartWalletFeatureEnabled;
-    const visibleAccounts = smartWalletFeatureEnabled
-      ? accounts
-      : accounts.filter(({ type }) => type !== ACCOUNT_TYPES.SMART_WALLET);
-
-    const visibleAccountsWithBalance = visibleAccounts.map((acc) => {
-      const accountBalances = balances[acc.id] || {};
-      const balance = calculatePortfolioBalance(assets, rates, accountBalances);
-      return { ...acc, balance };
+
+    const visibleAccounts = this.visibleAccounts(accounts, smartWalletFeatureEnabled);
+    const hasAccount = hasSmartWalletAccount(accounts);
```

Comment:
Please use `userHasSmartWallet` method instead. This one seems not to work. And this results in constant smart wallet init button. We only need to show it for those users who does not have smart wallet